public/javascript/index.js

```diff
@@ -1,23 +1,99 @@
-const charactersAPI = new APIHandler('http://localhost:8000');
+const charactersAPI = new APIHandler('https://minions-api.herokuapp.com/');
 
 window.addEventListener('load', () => {
   document.getElementById('fetch-all').addEventListener('click', function (event) {
+    
+    charactersAPI
+      .getFullList()
+      .then(response =>{
+        console.log(response)
+        const { data } = response
+        let characterList = ''
+        data.forEach(elm => {
+        characterList +=  `<div class="character-info"> <li> ${elm.name} <br> ${elm.occupation}<br> ${elm.weapon}</li></div>`
+        })
+        document.querySelector('.characters-container').innerHTML = characterList
+      })
+      .catch(err => console.log('error', err))
 
   });
 
   document.getElementById('fetch-one').addEventListener('click', function (event) {
 
+    const characterId = document.querySelector('#searchById').value
+
+     charactersAPI
+      .getOneRegister(characterId)
+      .then(oneCharacter =>{
+        console.log('este el character', oneCharacter) //dos horas atascado aqui y todo por que ponia numeros que habian borrado mis compañeros hasta que he caido... 
+
+        const character = oneCharacter.data
+
+        const characterList = `<div class="character-info"> 
+        <li>  name: ${character.name} 
+        <br> occupation: ${character.occupation}
+        <br> weapon: ${character.weapon}</li>
+        </div>`
+        console.log(characterList)
+        
+        document.querySelector('.characters-container').innerHTML = characterList
+      })
+      .catch(err => console.log('error', err))
+
   });
 
   document.getElementById('delete-one').addEventListener('click', function (event) {
 
+    const characterIdDel = document.querySelector('#deleteById').value
+
+    charactersAPI
+      .deleteOneRegister(characterIdDel)
+      .then(() => {
+        window.alert('borrado con exito')
+      })
+      .catch(err => console.log('error', err))
+
   });
 
   document.getElementById('edit-character-form').addEventListener('submit', function (event) {
+    event.preventDefault()
 
+    const editInputs = document.querySelectorAll('#edit-character-form input')
+
+    console.log(editInputs)
+
+    const editCharacter = {
+      name: editInputs[1].value,
+      occupation: editInputs[2].value,
+      weapon: editInputs[3].value
+    }
```

Comment:
Tanto en el edit como en el create te falta el isCartoon
```suggestion
    const editCharacter = {
      name: editInputs[1].value,
      occupation: editInputs[2].value,
      weapon: editInputs[3].value
      isCartoon: editInputs[4].checked
    }
```